### ISSUE TITLE
Dying out of nowhere is not welcoming for new players

### ISSUE THREAD

**Issue by Kalhama** [SoftbearStudios/mk48#188]
# Dying out of nowhere is not welcoming for new players
For new players it's hard to understand the game if they randomly die. Some actions which could alleviate this:
1. #180 
2. Making small ships not attractive targets for larger ships. (This could be contradicting that small ships are often great counters for larger ships which promotes rotation of large ships.)
3. Increasing radar range for smaller ships so they could avoid larger ships or at least see what killed them.
4. Making areas of map which promote being a small ship and discourage larger ships. Perhaps area of small islands or #129 which only smaller boats could access.

**Comment by finnbear** [comment]
Thanks for commenting! When it comes to this being a problem, you are exactly right. In fact, it is currently the game's largest problem, accounting for over half of the negative reviews we get on game sites (based on comments accompanying such reviews).

I tend to favor solution #4, possibly combined with solution #1 and #2 (I take issue with #3, since it compromises reality and ability to focus on one's ship). In particular, we hope to add a tropical/shallow area in the southern/bottom area of the map, where tightly packed islands and shallow water all but prohibit larger vessels (except littoral combat ships, i.e. Freedom, which are otherwise not as special as they should be)

**Comment by mitm001** [comment]
A good way of fixing this has been implemented by many games. Its real simple. 

You segment the map by level. The farther you travel north or whatever direction you choose the higher the level of interaction between players.

It starts out maybe level 1 area, then up it by two thereafter the further you travel. 

1
1-3
1-5
1-7
1-9

You then drop players into the highest level they can attain when res-pawning to prevent this problem. You can block players from traveling or firing into lower level areas based on max level ship they can attain.

This is how Runescape did it for their pvp area called the wilderness and it worked well. You can add a notification number that alerts the player to the area level they are in. When they first started they had similar problems with people hanging around spawn areas for newbies and slaughtering them. 

Your map is to small as it is so this would give you a chance to enlarge it.

**Comment by finnbear** [comment]
> You then drop players into the highest level they can attain when res-pawning to prevent this problem.

We did this a few months ago, but messed up the math pretty badly (y-coordinate was linear with respect to points, not level, causing a massive congregation in the south of the map). That led us to revert the feature, but we are pretty sure we will add it back in the next update (hopefully along with a new biome that is favorable to smaller ships).

TL;DR: It's a great idea, that just needs a slightly better implementation this time!

(yes, we are finally working on the next update! progress: 32 issues completed out of 61 total :tada:)

**Comment by WallKey1** [comment]
Ye, the draft of ship seems overlooked. Like a Kirov Draft, 9.1 m can travel really close to land, in reality not possible without rupturing hull. Should be shallower areas imo which could only be accessed if they were dredged out like a harbour. 

Fletcher-class destroyer ; Draft, 17.5 ft (5.3 m)
Visby-class_corvette Draught: 2.4 m (7.9 ft)

May start a new thingy for this, but do you think it is right that you can ram subs buy upgrading under them and vise versa. I know I use this tactic but it seems quite unfair because it is a get out of jail zero skill approach to killing an opponent. Like 1 decoy and upgrade to bismarak and kill someone who has earned 10k as a seawolf even if you kill yourself in the process. WDUT I think it is quite lame Even tho I use it as you would use every tactic within the physics of the game to survive. 


tks, 
k

**Comment by mitm001** [comment]
>May start a new thingy for this, but do you think it is right that you can ram subs buy upgrading under them 

The issue has been submitted already, #177. I agree with your statements on this.

**Comment by finnbear** [comment]
> Ye, the draft of ship seems overlooked. Like a Kirov Draft, 9.1 m can travel really close to land, in reality not possible without rupturing hull. Should be shallower areas imo which could only be accessed if they were dredged out like a harbour.

You're absolutely right. Well actually the code is 100% there (from last time we had shallow water), but the terrain currently drops off to 200m+ deep really fast.

> May start a new thingy for this, but do you think it is right that you can ram subs buy upgrading under them

One thing we can guarantee will be part of the next update is preventing surface ships from firing **any** weapon while rising up from underwater, whereas currently they can fire missiles and a few other types of weapons.

**Comment by FedorLap2006** [comment]
Want to give my feedback as well.
I also agree with solution 4. However I think additionally would be really good to make small ships harder to notice, from a distance. But since this might encourage small ships to try to kill bigger ships, small ships should be still noticable, but around the distance in which the guns of the bigger ship can reach it (so the ship won't be visible only in "point blank" zone).

**Comment by WallKey1** [comment]
@finnbear 

I had some time spare yesterday so put a bit of effort in. Resulting was 43978 points. 

So I quickly edited the screen record as a gift for your lol, really annoyingly I died from **lag** you can see in the last few moments like 10 players instantly leave went all strange, jumpy and disconnected O-o. SO ANNOYED!.

Hope you like the name I choose for my ship :D 🐿


Anyway it shows someone trying to kill me by turning into a clem one time amongst many, and when I upgraded to avoid death! And a few other bits I thought were funny within 4 hours of screen record.... I swear to god decoys don't always work esp with the Visby.  One time someone chased me then tried upgrading to Essex under ice to splat me. 😑

If it wasn't for the launching so many torps I would have been killed by that upgrading clem which relates to my point above. One tactic people use to avoid getting rammed up a sub upgrading to a ship is to surface but often they just end up beached on the deck of the ship😂 

Anyway, here is the link to the edit- (not great quality as don't have time for fancy) Just so you don't think I'm just all talk. 


https://youtu.be/HSK9_NO27DM



Tks, 
k

**Comment by mitm001** [comment]
> I swear to god decoys don't always work esp with the Visby.

I know they don't always work for a fact as I have witnessed multiple torpedoes in a group go by a decoy and sink me on many occasions. The decoy was not to close to the sub, positioned perfectly, and it was still moving away after they passed by. It happens enough that you know you cant rely on them.

I have also seen other decoys launched by opponents that were like freaking super magnets. Way to far out of range to be effective yet every torpedo would move to it no matter how long you delay firing.

I have also witnessed 3 out of 4 missiles fired from the yasen hit a ship and one go around the ship to hit something off screen somewhere. When I say go around I mean it literally looked like it went in a U when it was near the ship to go around it and return to the same path on other side to avoid hitting it.

Similar things happen with torpedo target selection where for whatever reason they will target a object behind the closest target. In those circumstances the torpedoes just move under the first target. 

Off topic, it would be desirable to selectively target with weapons deployment. To be able to pick out a target surrounded by others would be a great improvement.

**Comment by finnbear** [comment]
> I had some time spare so put a bit of effort in. Resulting was 43978 points.

Congratulations!

> really annoyingly I died from lag you can see in the last few moments like 10 players instantly leave went all strange, jumpy and disconnected O-o. SO ANNOYED!.

I apologize for the issues with server 1 yesterday. In the hours during and after the issue, we investigated and confirmed that the root cause was CPU starvation (the mk48.io servers run on shared-CPU VPS, so other tenants using too much CPU can cause lag in mk48.io). According to the support team for our VPS provider, server 1 was running on hardware with abnormally high load due (presumably due to multiple other tenants). We want to learn from, and hopefully prevent future instances of, this issue.

1) We are adding two new metrics to our internal statistics: 1) CPU steal i.e. percentage of CPU used by other VPS tenants 2) Seconds per server tick max/min i.e. to measure lag better
2) We are _considering_ renting dedicated CPU instead of shared CPU VPS's, although the cost is 6 times higher and the service isn't any better 99.9% of the time (the 0.1% is what happened yesterday)
3) We are prioritizing a system for transferring players between servers while preserving points (and ideally everything else)

> Hope you like the name I choose for my ship

:chipmunk: :chipmunk: :chipmunk: 

> I swear to god decoys don't always work esp with the Visby

You mean against Visby's torpedoes? Not sure why that would be...

> One time someone chased me then tried upgrading to Essex under ice to splat me

:100:

> Just so you don't think I'm just all talk.

I was already pretty impressed that you cataloged deadly ice :eyes:   (which will repel you in the next update)

**Comment by finnbear** [comment]
> I know they don't always work for a fact as I have witnessed multiple torpedoes in a group go by a decoy and sink me on many occasions. The decoy was not to close to the sub, positioned perfectly, and it was still moving away after they passed by. It happens enough that you know you cant rely on them.
>
> I have also seen other decoys launched by opponents that were like freaking super magnets. Way to far out of range to be effective yet every torpedo would move to it no matter how long you delay firing.

A big factor with how well decoys work is whether you don't have active sensors on. Also, ideally you are a small sub with low sonar cross-section, not Ohio.

Next update changes weapon guidance a bit. Not necessarily in the exact ways you suggest, but we think its an improvement.

**Comment by WallKey1** [comment]
Had to happen when I was playing seriously and not just to mess about 😂

So hang on decoys work better when you have sensors on🤔

**Comment by finnbear** [comment]
> Had to happen when I was playing seriously and not just to mess about :joy:

Yeah, yesterday was an example of "if it can go wrong, it will go wrong, and at the worst time"

At least the failover system kinda worked to route incoming players to server 2 when server 1 became unhealthy. Incidents like this help us improve this system even further :smile: 

> So hang on decoys work better when you have sensors onthinking

Actually they work much better when you have your sensors (sonar) **off**. That's because active sonar makes noise that attracts torpedoes towards you (and not the decoy)

**Comment by finnbear** [comment]
> This game is genuinely harsh on new players

We completely agree and want to fix it!

> it it is worthy of proper investigation as mentioned.

While not necessarily an investigation, we closely monitor aggregate statistics.

> I'm not sure what your player turnover rate is..

At any given time, around ~40% of players are new (~60% are returning to the game). We think there is significant room for improvement here.

> maybe it would benefit from a introductory tutorial (skippable) where you were unable to kill anything or be attacked (par bots)

Yeah this could help. But with the situation as bad as it is now, there are even simpler improvements we could make and some that we plan to release in the next update.

> Hope you figure out how to monetize or use as leverage for other projects

Yes, this is a medium-term to long-term concern ;)

We are working on more games (although they are temporarily on hold due to the continued development of mk48.io), and do hope to monetize them to make it sustainable.

**Comment by finnbear** [comment]
Situation has been improved in latest update :)

**Comment by mitm001** [comment]
> Situation has been improved in latest update :)

Not really, you just moved the level of attacker down slightly. 

You still get spawn killed well over 20 times before you gain traction with a higher level ship. Assuming your experienced in the first place.

I will say, it can be fun sparring with lower levels in the south though. Making most islands missile proof would help more than anything else.

**Comment by mitm001** [comment]
I have simple suggestion for the level 1 class of ships. Give all of them the same range for visual and make it similar to the ohio range.

Theirs no real advantage to a level one having this except it allows the player to at least see who just sank them.

**Comment by XenonPy** [comment]
I have an idea for how to solve this, without changing any ships. The main reason people kill ships is to level up. My thought process it that something like this would decrease the motivation:
```
xp gained = ship killed base xp - (x^2)/8 + y
```
y would be adjusted as a fixed increase, as I don't know the scale exactly. Not sure if there is already something like this, or if it would work.

**Comment by finnbear** [comment]
It's unclear what `y` is in your formula, but we already implemented a system that reduces points for killing ships that are lower level than you.

**Comment by XenonPy** [comment]
y is a fixed increase, x is the base value of xp. If there is already a system like this then no need
